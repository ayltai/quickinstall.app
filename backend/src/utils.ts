import { ErrorReporting, } from '@google-cloud/error-reporting';

const errors = new ErrorReporting();

export const handleError = (error : any) => {
    if (process.env.NODE_ENV === 'production') {
        errors.report(error);
    } else {
        console.error(error);
    }
};
