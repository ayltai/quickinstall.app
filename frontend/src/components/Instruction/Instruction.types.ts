import type { DialogViewProps, } from '../DialogView';

export interface InstructionProps extends DialogViewProps {
    transactionId? : string,
}
