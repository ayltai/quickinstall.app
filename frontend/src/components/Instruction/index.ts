export { Instruction, } from './Instruction';
export type { InstructionProps, } from './Instruction.types';
