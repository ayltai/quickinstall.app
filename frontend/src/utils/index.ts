import * as Sentry from '@sentry/react';
import { FirebaseApp, } from 'firebase/app';
import { getAnalytics, logEvent, } from 'firebase/analytics';

export const handleError = (error : any) => {
    if (import.meta.env.PROD) {
        Sentry.captureException(error);
    } else {
        console.error(error);
    }
};

export const logPageView = (app : FirebaseApp, {
    pageName,
    route,
} : {
    pageName : string,
    route    : string,
}) => {
    try {
        logEvent(getAnalytics(app), 'page_view', {
            page_title    : pageName,
            page_location : route,
        });
    } catch (error) {
        handleError(error);
    }
};

export const logAddToCart = (app : FirebaseApp, {
    platform,
    itemId,
} : {
    platform : 'win' | 'mac',
    itemId   : string,
}) => {
    try {
        logEvent(getAnalytics(app), 'add_to_cart', {
            items : [
                {
                    item_id : `${platform}/${itemId}`,
                },
            ],
        });
    } catch (error) {
        handleError(error);
    }
};

export const logRemoveFromCart = (app : FirebaseApp, {
    platform,
    itemId,
} : {
    platform : 'win' | 'mac',
    itemId   : string,
}) => {
    try {
        logEvent(getAnalytics(app), 'remove_from_cart', {
            items : [
                {
                    item_id : `${platform}/${itemId}`,
                },
            ],
        });
    } catch (error) {
        handleError(error);
    }
};

export const logCheckout = (app : FirebaseApp, {
    platform,
    itemIds,
} : {
    platform : 'win' | 'mac',
    itemIds  : string[],
}) => {
    try {
        logEvent(getAnalytics(app), 'checkout', {
            items : itemIds.map((itemId : string) => ({
                item_id : `${platform}/${itemId}`,
            })),
        });
    } catch (error) {
        handleError(error);
    }
};

export const logSearch = (app : FirebaseApp, {
    platform,
    query,
} : {
    platform : 'win' | 'mac',
    query    : string,
}) => {
    try {
        logEvent(getAnalytics(app), 'search', {
            search_term : `${platform}/${query}`,
        });
    } catch (error) {
        handleError(error);
    }
};

export const logShare = (app : FirebaseApp, {
    product,
    itemId,
} : {
    product : 'win/packages' | 'mac/packages' | 'tweaks',
    itemId  : string,
}) => {
    try {
        logEvent(getAnalytics(app), 'share', {
            content_type : product,
            item_id      : itemId,
        });
    } catch (error) {
        handleError(error);
    }
};
