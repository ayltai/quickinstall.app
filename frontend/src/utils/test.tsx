import { createTheme, ThemeProvider, } from '@mui/material';
import { configureStore, type Store, } from '@reduxjs/toolkit';
import { render, } from '@testing-library/react';
import React, { type ReactNode, } from 'react';
import { Provider, } from 'react-redux';
import { MemoryRouter, } from 'react-router-dom';

import { appStoreApi, chocolateyApi, contentfulApi, homebrewApi, quickInstallApi, } from '../apis';
import { recipeReducer, preferenceReducer, spellReducer, } from '../states';
import { appTheme, } from '../styles';

export const createStore = (preloadedState? : unknown) => configureStore({
    preloadedState,
    reducer    : {
        recipe                          : recipeReducer,
        preference                      : preferenceReducer,
        spell                           : spellReducer,
        [ appStoreApi.reducerPath  ]    : appStoreApi.reducer,
        [ chocolateyApi.reducerPath ]   : chocolateyApi.reducer,
        [ contentfulApi.reducerPath ]   : contentfulApi.reducer,
        [ homebrewApi.reducerPath  ]    : homebrewApi.reducer,
        [ quickInstallApi.reducerPath ] : quickInstallApi.reducer,
    },
    middleware : getDefaultMiddleware => getDefaultMiddleware().concat(appStoreApi.middleware, chocolateyApi.middleware, contentfulApi.middleware, homebrewApi.middleware, quickInstallApi.middleware),
});

export const defaultStore = createStore();

export type RootState = ReturnType<typeof defaultStore.getState>;

// @ts-ignore
const theme = createTheme(appTheme);

const customRender = (ui : any, {
    preloadedState,
    store = createStore(preloadedState),
    ...rest
} : {
    preloadedState?   : Record<string, any>,
    store?            : Store<RootState>,
    [ rest : string ] : any,
} = {}) => render(ui, {
    wrapper : ({
        children,
    } : {
        children : ReactNode,
    }) => (
        <Provider store={store}>
            <MemoryRouter>
                <ThemeProvider theme={theme}>
                    {children}
                </ThemeProvider>
            </MemoryRouter>
        </Provider>
    ),
    ...rest,
});

export * from '@testing-library/react';

export { customRender as render, };
