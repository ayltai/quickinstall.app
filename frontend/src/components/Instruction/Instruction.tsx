import Stack from '@mui/material/Stack';
import Typography from '@mui/material/Typography';
import Link from '@mui/material/Link';
import React, { type FC, useCallback, useContext, } from 'react';
import { useTranslation, } from 'react-i18next';

import { API_ENDPOINT, } from '../../constants';
import { FirebaseContext, } from '../../contexts';
import { useAppSelector, } from '../../hooks';
import { logShare, } from '../../utils';
import { CopyableInput, } from '../CopyableInput';
import { DialogView, } from '../DialogView';
import type { InstructionProps, } from './Instruction.types';

/**
 * An instruction dialog that displays installation instructions to the user.
 * @param transactionId The ID of the transaction
 * @param props Other props
 */
export const Instruction : FC<InstructionProps> = ({
    transactionId,
    ...props
}) => {
    const { open, title, color, children, ...rest } = props;

    const { product, } : { product : 'win/packages' | 'mac/packages' | 'tweaks', } = useAppSelector(state => state.preference);

    const app = useContext(FirebaseContext);

    const { t, } = useTranslation();

    const handleCopy = useCallback(() => {
        if (app && transactionId) logShare(app, {
            product,
            itemId  : transactionId,
        });
    }, [ app, transactionId, product, ]);

    return (
        <DialogView
            {...rest}
            color={color}
            open={open && !!transactionId}
            title={t('checkout.title')}>
            <Stack spacing={1}>
                {children}
                <Typography>
                    {title}
                </Typography>
                {product === 'win/packages' && (
                    <Link
                        color={color === 'info' ? 'info.main' : color}
                        href='https://www.makeuseof.com/windows-run-command-prompt-admin/'
                        target='_blank'
                        rel='noreferrer'>
                        https://www.makeuseof.com/windows-run-command-prompt-admin/
                    </Link>
                )}
                {product === 'mac/packages' && (
                    <Link
                        color={color === 'info' ? 'info.main' : color}
                        href='https://support.apple.com/en-gb/guide/terminal/apd5265185d-f365-44cb-8b09-71a064a42125/mac'
                        target='_blank'
                        rel='noreferrer'>
                        https://support.apple.com/en-gb/guide/terminal/apd5265185d-f365-44cb-8b09-71a064a42125/mac
                    </Link>
                )}
                <CopyableInput
                    value={product === 'win/packages' ? `powershell -command "Invoke-WebRequest ${import.meta.env.DEV ? `http://${window.location.hostname}:8080` : API_ENDPOINT}/win/recipes/${transactionId}/install -OutFile quickinstall.bat" && quickinstall.bat && del quickinstall.bat` : `bash -c "$(curl -fsSL '${import.meta.env.DEV ? `http://${window.location.hostname}:8080` : API_ENDPOINT}/${product === 'mac/packages' ? 'mac/recipes' : 'mac/spells'}/${transactionId}/install')"`}
                    onCopy={handleCopy} />
            </Stack>
        </DialogView>
    );
};
