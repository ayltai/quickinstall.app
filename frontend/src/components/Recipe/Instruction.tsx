import Box from '@mui/material/Box';
import Link from '@mui/material/Link';
import Typography from '@mui/material/Typography';
import React, { type FC, } from 'react';
import { useTranslation, } from 'react-i18next';

import { useAppSelector, } from '../../hooks';
import type { Ingredient, } from '../../models';
import { Instruction, type InstructionProps, } from '../Instruction';

/**
 * An instruction dialog that informs users about how to install the selected packages.
 */
export const RecipeInstruction : FC<InstructionProps> = props => {
    const { transactionId, ...rest } = props;

    const { ingredients, } : { ingredients : Ingredient[],     } = useAppSelector(state => state.recipe);

    const { product, themeMode,   } : { product : 'win/packages' | 'mac/packages' | 'tweaks', themeMode   : 'light' | 'dark', } = useAppSelector(state => state.preference);

    const { t, } = useTranslation();

    return (
        <Instruction
            transactionId={transactionId}
            title={t(product === 'win/packages' ? 'packages.checkout.copy.win' : 'packages.checkout.copy.mac')}
            {...rest}>
            {ingredients.some(ingredient => ingredient.source === 'App Store') && (
                <>
                    <Typography>
                        {t('packages.checkout.instructions')}
                    </Typography>
                    <Link
                        color='primary'
                        href='https://support.apple.com/en-gb/guide/app-store/fir6253293d/mac'
                        target='_blank'
                        rel='noreferrer'>
                        https://support.apple.com/en-gb/guide/app-store/fir6253293d/mac
                    </Link>
                    <Box
                        sx={{
                            borderRadius    : '1rem',
                            backgroundColor : themeMode === 'light' ? '#dcedc8' : '#558b2f',
                        }}
                        padding={2}>
                        {ingredients.filter(ingredient => ingredient.source === 'App Store').map(ingredient => (
                            <Typography key={ingredient.id}>
                                {`â€¢ ${ingredient.name}`}
                            </Typography>
                        ))}
                    </Box>
                </>
            )}
        </Instruction>
    );
};
