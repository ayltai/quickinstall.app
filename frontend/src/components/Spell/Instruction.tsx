import Box from '@mui/material/Box';
import Link from '@mui/material/Link';
import Typography from '@mui/material/Typography';
import React, { type FC, } from 'react';
import { useTranslation, } from 'react-i18next';

import { useAppSelector, } from '../../hooks';
import type { Tweak, } from '../../models';
import { Instruction, InstructionProps, } from '../Instruction';

/**
 * An instruction dialog that informs users about how to apply the selected tweaks.
 */
export const SpellInstruction : FC<InstructionProps> = props => {
    const { transactionId, ...rest } = props;

    const { tweaks,    } : { tweaks    : Tweak[],          } = useAppSelector(state => state.spell);
    const { themeMode, } : { themeMode : 'light' | 'dark', } = useAppSelector(state => state.preference);

    const { t, } = useTranslation();

    return (
        <Instruction
            color='secondary'
            transactionId={transactionId}
            title={t('tweaks.checkout.copy')}
            {...rest}>
            {tweaks.some(tweak => tweak.sipInvolved) && (
                <>
                    <Box paddingBottom={2}>
                        <Typography>
                            {t('tweaks.checkout.instructions.1')}
                        </Typography>
                        <Box
                            sx={{
                                borderRadius    : '1rem',
                                backgroundColor : themeMode === 'light' ? '#ffccbc' : '#d84315',
                            }}
                            padding={2}>
                            {tweaks.filter(tweak => tweak.sipInvolved).map(tweak => (
                                <Typography key={tweak.id}>
                                    {`â€¢ ${tweak.name}`}
                                </Typography>
                            ))}
                        </Box>
                    </Box>
                    <Box paddingBottom={2}>
                        <Typography>
                            {t('tweaks.checkout.instructions.2')}
                        </Typography>
                        <Link
                            color='secondary'
                            href='https://developer.apple.com/documentation/security/disabling_and_enabling_system_integrity_protection'
                            target='_blank'
                            rel='noreferrer'>
                            https://developer.apple.com/documentation/security/disabling_and_enabling_system_integrity_protection
                        </Link>
                    </Box>
                </>
            )}
        </Instruction>
    );
};
