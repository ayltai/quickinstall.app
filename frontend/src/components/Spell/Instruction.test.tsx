import React from 'react';

import { render, } from '../../utils/test';
import { SpellInstruction, } from './Instruction';

describe('<Instruction />', () => {
    it('renders correctly', () => expect(render(
        <SpellInstruction
            open={true}
            transactionId='dummy' />
    )).toMatchSnapshot());

    it('renders instructions correctly', () => expect(render(
        <SpellInstruction
            open={true}
            transactionId='dummy' />,
        {
            spell : {
                tweaks : [
                    {
                        id         : '1',
                        name       : 'Magic',
                        parameters : [
                            'Disable',
                        ],
                        types      : [
                            'boolean',
                        ],
                        values     : [
                            true,
                        ],
                    },
                ],
            },
        }
    )).toMatchSnapshot());
});
