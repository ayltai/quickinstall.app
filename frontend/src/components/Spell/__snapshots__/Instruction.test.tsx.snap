// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Instruction /> > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-1sep8xo-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby=":r0:"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-7xpjjc-MuiPaper-root-MuiDialog-paper"
          role="dialog"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-14o58l0-MuiTypography-root-MuiDialogTitle-root"
            id=":r0:"
          >
            checkout.title
          </h2>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-lc2yqr-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiDialogContent-root MuiDialogContent-dividers css-1t4vnk2-MuiDialogContent-root"
          >
            <div
              class="MuiStack-root css-jfdv4h-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1che66z-MuiTypography-root"
              >
                tweaks.checkout.copy
              </p>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1g8o8w2-MuiTypography-root-MuiLink-root"
                href="https://support.apple.com/en-gb/guide/terminal/apd5265185d-f365-44cb-8b09-71a064a42125/mac"
                rel="noreferrer"
                target="_blank"
              >
                https://support.apple.com/en-gb/guide/terminal/apd5265185d-f365-44cb-8b09-71a064a42125/mac
              </a>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary css-1n0mhtb-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  type="text"
                  value="bash -c "$(curl -fsSL 'http://localhost:8080/mac/recipes/dummy/install')""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      â€‹
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiDialogActions-root MuiDialogActions-spacing css-knqc4i-MuiDialogActions-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorSecondary css-d0whyw-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              action.close
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Instruction /> > renders instructions correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-1sep8xo-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby=":r1:"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-7xpjjc-MuiPaper-root-MuiDialog-paper"
          role="dialog"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-14o58l0-MuiTypography-root-MuiDialogTitle-root"
            id=":r1:"
          >
            checkout.title
          </h2>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-lc2yqr-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiDialogContent-root MuiDialogContent-dividers css-1t4vnk2-MuiDialogContent-root"
          >
            <div
              class="MuiStack-root css-jfdv4h-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1che66z-MuiTypography-root"
              >
                tweaks.checkout.copy
              </p>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1g8o8w2-MuiTypography-root-MuiLink-root"
                href="https://support.apple.com/en-gb/guide/terminal/apd5265185d-f365-44cb-8b09-71a064a42125/mac"
                rel="noreferrer"
                target="_blank"
              >
                https://support.apple.com/en-gb/guide/terminal/apd5265185d-f365-44cb-8b09-71a064a42125/mac
              </a>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary css-1n0mhtb-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  type="text"
                  value="bash -c "$(curl -fsSL 'http://localhost:8080/mac/recipes/dummy/install')""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      â€‹
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiDialogActions-root MuiDialogActions-spacing css-knqc4i-MuiDialogActions-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorSecondary css-d0whyw-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              action.close
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
